### ISSUE TITLE
Is there a way to give a `GeoJSONLayer` a name?

### ISSUE THREAD

**Issue by vomc** [alex3165/react-mapbox-gl#804]
I am rendering some geoJson polygons in my app using the `GeoJSONLayer` component. I have provided some `fill` expressions and things are working well.

I noticed this layer is by default called `geojson-1-fill` and it seems that neither `Layer` nor the `GeoJSONLayer` component allow you to change this. 

Is there a way to specifically give that layer a name?

**Comment by mklopets** [comment]
Layer has an id prop and geojsonlayer has layerOptions

**Comment by vomc** [comment]
Ah ok. So I did `layerOptions={{id: 'dskjsdjk823uj3jbkjxiugd87s87gsiuk'}}` but now I am getting: 

```index.js:1437 Error: Layer with id "dskjsdjk823uj3jbkjxiugd87s87gsiuk" already exists on this map
    at i.addLayer (mapbox-gl.js:19037)
    at r.addLayer (mapbox-gl.js:22876)```

Is it somehow adding this layer over and over again?

**Comment by mklopets** [comment]
there might be a bug when it rerenders the GeoJSON... are you having the component rerender?

**Comment by vomc** [comment]
Yes, I do. That is I am rendering something that is a changing collection of GeoJson entities. So this error is throughout the app if I do assign a layer name ... 🤔 

Is there a way around it? I guess all I need is to set a specific layer so that I can use `queryRenderedFeatures` and restrict the search to a specific layer...

**Comment by solarstar101** [comment]
any update?
